### ISSUE TITLE
Import stopped in osm2pgsql step

### ISSUE THREAD

**Issue by issahammoud** [openstreetmap/Nominatim#1566]
Hy,

I tried to install Nominatim project several times on a fresh Ubuntu, 16.04 LTS, amd64 xenial image on Google Cloud.

I followed [this tutorial](https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-16/) and It is not the first time I installed the project based on this tutorial and it works well before.<br>
I followed it step by step, with just fixing a typo (the folder is called Nominatim-3.4.0, but in nominatim.conf they forgot the version number and used only Nominatim).<br>

It is worthing to mention that the installation took a lot of time (about 10 hours with 32 GB RAM used and 8 CPU).<br>
When it is done I tried to access it through "http://myexternalIP/nominatim". I adjusted all Apache2 configuration and I was able to get Apache welcome page.<br>

So, when I tried to access "http://myexternalIP/nominatim/?debug=1" I got __500 Internal Server Error__ and the following details:

___
Nominatim has encountered an internal error while accessing the database. This may happen because the database is broken or because of a bug in the software.

Details
Database query failed
Exception Nominatim\DatabaseError thrown in /srv/nominatim/Nominatim/lib/DB.php(187).

SQL Error
SQLSTATE[42P01]: Undefined table: 7 ERROR: relation "import_status" does not exist LINE 1: ...lastimportdate,'YYYY/MM/DD HH24:MI')||' GMT' from import_sta... ^
PDOException::__set_state(array(
   'message' => 'SQLSTATE[42P01]: Undefined table: 7 ERROR:  relation "import_status" does not exist
LINE 1: ...lastimportdate,\'YYYY/MM/DD HH24:MI\')||\' GMT\' from import_sta...
                                                             ^',
   'string' => '',
   'code' => '42P01',
   'file' => '/srv/nominatim/Nominatim/lib/DB.php',
   'line' => 184,
   'trace' => 
  array (
    0 => 
    array (
      'file' => '/srv/nominatim/Nominatim/lib/DB.php',
      'line' => 184,
      'function' => 'query',
      'class' => 'PDO',
      'type' => '->',
      'args' => 
      array (
        0 => 'select TO_CHAR(lastimportdate,\'YYYY/MM/DD HH24:MI\')||\' GMT\' from import_status limit 1',
      ),
    ),
    1 => 
    array (
      'file' => '/srv/nominatim/Nominatim/lib/DB.php',
      'line' => 96,
      'function' => 'getQueryStatement',
      'class' => 'Nominatim\\DB',
      'type' => '->',
      'args' => 
      array (
        0 => 'select TO_CHAR(lastimportdate,\'YYYY/MM/DD HH24:MI\')||\' GMT\' from import_status limit 1',
        1 => NULL,
        2 => 'Database query failed',
      ),
    ),
    2 => 
    array (
      'file' => '/srv/nominatim/Nominatim/website/search.php',
      'line' => 70,
      'function' => 'getOne',
      'class' => 'Nominatim\\DB',
      'type' => '->',
      'args' => 
      array (
        0 => 'select TO_CHAR(lastimportdate,\'YYYY/MM/DD HH24:MI\')||\' GMT\' from import_status limit 1',
      ),
    ),
    3 => 
    array (
      'file' => '/srv/nominatim/Nominatim/build/website/search.php',
      'line' => 3,
      'args' => 
      array (
        0 => '/srv/nominatim/Nominatim/website/search.php',
      ),
      'function' => 'require_once',
    ),
  ),
   'previous' => NULL,
   'errorInfo' => 
  array (
    0 => '42P01',
    1 => 7,
    2 => 'ERROR:  relation "import_status" does not exist
LINE 1: ...lastimportdate,\'YYYY/MM/DD HH24:MI\')||\' GMT\' from import_sta...
                                                             ^',
  ),
))
Stack trace
#0 /srv/nominatim/Nominatim/lib/DB.php(96): Nominatim\DB->getQueryStatement('select TO_CHAR(...', NULL, 'Database query ...')
#1 /srv/nominatim/Nominatim/website/search.php(70): Nominatim\DB->getOne('select TO_CHAR(...')
#2 /srv/nominatim/Nominatim/build/website/search.php(3): require_once('/srv/nominatim/...')
#3 {main}
If you feel this error is incorrect feel file an issue on Github. Please include the error message above and the URL you used.
___
Then I looked at `/var/log/syslog` file and I found some lines containing the following errors:
___
Nov 12 06:02:33 nominatim systemd-udevd[568]: error opening ATTR{/sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/sda1/queue/scheduler} for writing: No such file or directory
Nov 12 06:02:33 nominatim systemd-udevd[1265]: error opening ATTR{/sys/devices/pci0000:00/0000:00:03.0/virtio0/host0/target0:0:1/0:0:1:0/block/sda/sda1/queue/scheduler} for writing: No such file or directory
Nov 12 06:02:33 nominatim kernel: [    0.965662] RAS: Correctable Errors collector initialized.
___
So I checked if this file exists or not, and I found that the path ends at `/sys/devices/pci0000\:00/0000\:00\:03.0/virtio0/host0/target0\:0\:1/0\:0\:1\:0/block/sda/sda1/` and there is no `queue/scheduler`.

Anyone know how to fix it? I passed several days trying to let it work.

Thanks in advance and sorry for my poor english.

**Comment by mtmail** [comment]
Thanks for mentioning the operating system version, Nominatim version, import time and URLs you tried.

10 hours is still a good import time, with the same hardware and the full planet it would likely take a whole week. 

Do you have the last output lines of the import? In the tutorial (https://nominatim.org/release-docs/latest/admin/Import-and-Update/) that's called `setup.log`. Ideally it would say `Setup finished.` You 

Do queries from command-line work? There's a script in the `build` directory `utils/query.php` where you can run queries without the webbrowser.

In the browser `/nominatim` might not work, try `/nominatim/search.php`. Does it show a website with search form? Does `/nominatim/status.php` and `/nominatim/status.php?format=json` work?

When you connect to the database on command-line (with the `psql` command), what do the following queries return?
`select count(*) from places;`
`select indexed_status, count(*) from placex group by indexed_status;`
`select * from import_status;`

The lines from `syslog.log` are unrelated. What we usually look for is if the postgresql database crashed.

**Comment by issahammoud** [comment]
Thank you for your answer.
First of all, 10 hours was for installing the minimal version (import-admin.style).
About the logs in `setup.log` file, the last line is `Processing: Node(5568842k 6543.9k/s) Way(617440k 48.78k/s) Relation(1689200 77.62/s)`. It seems that it didn't finish but there is no error at the end of the file, even though I redirected the standard and error output to the file (2>&1).
I ran it in the background and when I returned there were no new logs written and no resources in use, so I supposed that it was done.
None of the suggested urls work (neither search.php nor status.php). They returned the same error, 500 internal server error.
Result of queries:
* select count(*) from places; --> ERROR:  relation "places" does not exist
* select * from import_status; --> ERROR:  relation "import_status" does not exist

**Comment by mtmail** [comment]
Ok, so the import didn't finish yet. At the `Processing: Node...` step osm2pgsql is running and creating database entries before the indexing starts. It should have finished with lines like
```
Node stats: total(5194866599), max(6468955701) in 1005s
Way stats: total(578627169), max(689562594) in 5770s
Relation stats: total(6745863), max(9584064) in 6332s
Setting up table: planet_osm_nodes
Setting up table: planet_osm_ways
Setting up table: planet_osm_rels
```

Usually either the database server crashed, the postgresql logs might be helpful (ignore any warnings about wal intervals). Or the process itself crashed due to memory limitation. If Google Cloud has monitoring check the RAM usage. Sadly I don't know how much the import-admin.style would need. osm2pgsql tends to crash without much warning on low memory.

If it's easy upgrade the machine to 64GB RAM, after the import you can lower it again.

A flatnode file https://nominatim.org/release-docs/latest/admin/Import-and-Update/#flatnode-files helps to keep memory usage low.

**Comment by mtmail** [comment]
Might also be related to https://github.com/openstreetmap/Nominatim/issues/1550

**Comment by issahammoud** [comment]
Thank you I will try to reinstall it.
I used flatnode, and I specified 32GB over 38GB in total for the task and there were nothing running at the same time.
It is worthing to mention that the first time I imported the data with this version of Nominatim, it worked for several minutes, then when I used Python Nominatim API from geopy for querying it stopped working with 500 status code. I really don't understand what happened.
Thank you I will try to reinstall it.
